src/lazy/expanded/macro_evaluator.rs

```diff
@@ -1058,6 +1084,58 @@ impl<'top, D: Decoder> MakeStringExpansion<'top, D> {
     }
 }
 
+/// The evaluation state of the `make_symbol` macro.
+///
+/// `(:make_symbol ...)` eagerly expands each of its arguments in turn, concatenating the resulting
+/// string and symbol values in order to make a single symbol.
+///
+/// This allows a writer to construct a symbol from fragments, some or all of which may reside
+/// in the symbol or macro tables.
+///
+/// If any of the arguments expand to a non-text value, `make_string` will return an error.
+///
+/// Examples:
+///   (:make_string "foo" "bar")              => foobar
+///   (:make_string foo bar)                  => foobar
+///   (:make_string "foo" bar)                => foobar
+///   (:make_string "first_" $4)              => first_name
+///   (:make_string (:values "first" "_") $4) => first_name
+///   (:make_string)                          => ''
+///   (:make_string "foo" 7)                  => Error
+#[derive(Copy, Clone, Debug)]
+pub struct MakeSymbolExpansion<'top, D: Decoder> {
+    arguments: MakeTextExpansion<'top, D>,
+}
+
+impl<'top, D: Decoder> MakeSymbolExpansion<'top, D> {
+    pub fn new(arguments: MacroExprArgsIterator<'top, D>) -> Self {
+        Self {
+            arguments: MakeTextExpansion::new(arguments),
+        }
+    }
+
+    /// Yields the next [`ValueExpr`] in this `make_symbol` macro's evaluation.
+    pub fn next(
+        &mut self,
+        context: EncodingContextRef<'top>,
+        _environment: Environment<'top, D>,
+    ) -> IonResult<MacroExpansionStep<'top, D>> {
+        let constructed_text = self.arguments.concatenate_text_arguments(context)?;
```

Comment:
🗺️ The new `make_symbol` impl calls `MakeTextExpansion::concatenate_text_arguments` and then massages the resulting `&'bump str` into a `ValueRef`.